### PR TITLE
Framework/port lib tree-convert to single runner

### PR DESCRIPTION
**Pull request by jhnstn**
part of #3942 

Port lib/tree-convert to single test runner and some ES6 refactoring.
Suite works in isolation with `.only`

**Testing**

`npm run test-client`

cc @gziolo

### CODE REVIEW

**Comment by jhnstn** [comment]
@gziolo  I _just_noticed that `test/tree-convert.js` should be `test/index.js`. I can open a follow up PR to fix that

**Comment by gziolo** [comment]
If test file is short you can rename in this PR :)

Dnia 05.04.2016 o godz. 18:22 Jason Johnston <notifications@github.com> napisał(a):

**Comment by gziolo** [comment]
This time commits history looks good :)

File: client/lib/tree-convert/test/tree-convert.js [hunk 1]
```diff
@@ -1,69 +1,73 @@
...
+			const treeified = converter.treeify( menu.items );
 
 			expect( treeified[1].items[0] ).not.to.have.property( 'order' );
 		} );
 
-		it( 'should handle bad parent data, falling back to parent 0', function() {
-			var menu = cloneDeep( fixtures.menuBadParent ),
-				treeified = this.converter.treeify( menu.items );
+		it( 'should handle bad parent data, falling back to parent 0', () => {
+			const badMenu = cloneDeep( menuBadParent );
```
Comment (gziolo) [inline client/lib/tree-convert/test/tree-convert.js]: I must have explain badly how we deal with `const`. `let` and `var` :)

We usually use one `const` declaration for all variables in given scope. Exception would the case when you want to use it after `if` statement or inside nested block like:
```js
const a = 1, b = 2, c = 3;
let i = 0;
while ( i < c ) {
   let valid = true;
   if (  i < b) {
     valid = false;
   }
   i = i +2;
}

for ( let j = 0; j < i; j++ ) {
  const sth = 'sth';
}
```
This code most likely doesn't make any sense. I just wanted to present the idea :)

File: client/lib/tree-convert/test/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		} );
+
+		it( 'should remove the "order" property from items, since order will be implicit', () => {
+			const treeified = converter.treeify( menu.items );
+
+			expect( treeified[1].items[0] ).not.to.have.property( 'order' );
+		} );
+
+		it( 'should handle bad parent data, falling back to parent 0', () => {
+			const badMenu = cloneDeep( menuBadParent );
```
Comment (gziolo) [inline client/lib/tree-convert/test/index.js]: I must have explain badly how we deal with `const`. `let` and `var` :)

We usually use one `const` declaration for all variables in given scope. Exception would the case when you want to use it after `if` statement or inside nested block like:
```js
const a = 1, b = 2, c = 3;
let i = 0;
while ( i < c ) {
   let valid = true;
   if (  i < b) {
     valid = false;
   }
   i = i +2;
}

for ( let j = 0; j < i; j++ ) {
  const sth = 'sth';
}
```
This code most likely doesn't make any sense. I just wanted to present the idea :)

File: client/lib/tree-convert/test/fixtures.js [hunk 3]
```diff
@@ -1,60 +1,58 @@
...
-				locations: [ 'secondary' ]
-			}
-		],
-		locations : [
-			{ name: 'primary', description: 'Top &amp; Primary Menu' },
-			{ name: 'secondary', description: 'Secondary menu in left sidebar' },
-			{ name: 'social&amp;lite', description: 'Social Menu' }
-		]
-	},
+export const menusFlat = {
```
Comment (gziolo) [inline client/lib/tree-convert/test/fixtures.js]: We should move this file to subfolder. New name should be like:
`client/lib/tree-convert/test/fixtures/index.js`

At the moment we use `glob` to find all test files inside `test` folders using this pattern `client/**/test/*.js` (see https://github.com/Automattic/wp-calypso/blob/master/circle.yml#L21). Circle CI tries to distribute all test files found across all containers (4 at the moment). That's why we should avoid putting all non-test files directly inside `test` folder to maintain nice test distribution and avoid loading unnecessary files.

File: client/lib/tree-convert/test/index.js (hunk 2 above)
Comment (jhnstn) [inline client/lib/tree-convert/test/index.js]: I don't think your prior explanation was off. _I think_ this is another habit of mine. We've been enforcing one declaration per line.

Just to make sure I got this, the correct `const` usage here would be: 
```
const badMenu = cloneDeep( menuBadParent ), treeified = converter.treeify( badMenu.items );
// or
const badMenu = cloneDeep( menuBadParent ) ,
  treeified = cloneDeep( badMenu.items );
```

And the same applies to `let` I'm assuming. Extending your example:

```
const a = 1, b = 2, c = 3;
let i = 0, j = 0; 
...

**Comment by lancewillett** [comment]
Sorry for the bot noise... my fault.

File: client/lib/tree-convert/test/index.js (hunk 2 above)
Comment (gziolo) [inline client/lib/tree-convert/test/index.js]:
```js
const badMenu = cloneDeep( menuBadParent ),
  treeified = cloneDeep( badMenu.items );
```
is the way to go here :)

I guess the most conventions friendly version would be:
```js
const a = 1,
    b = 2,
    c = 3;
let i, j, k,
    m = 4,
    n = 5;
```
Complicated topic :D

**Comment by gziolo** [comment]
Thanks for addressing my feedback, it looks great now. Merging :shipit:

**Pull request merged**